modules/460-log-shipper/hooks/migrate_cluster_log_destination.go

```
@@ -0,0 +1,136 @@
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "cluster_log_destination",
+			ApiVersion: "deckhouse.io/v1alpha1",
+			Kind:       "ClusterLogDestination",
+			FilterFunc: filterClusterLogDestination,
+		},
+		{
+			Name:       "token",
+			ApiVersion: "v1",
+			Kind:       "Secret",
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{MatchNames: []string{
+					"d8-log-shipper",
+				}},
+			},
+			NameSelector: &types.NameSelector{MatchNames: []string{
+				"log-shipper-token",
+			}},
+			FilterFunc: filterLogShipperTokenSecret,
+		},
+	},
+}, migrateClusterLogDestination)
+
+func migrateClusterLogDestination(input *go_hook.HookInput) error {
+	destinationSnapshots := input.Snapshots["cluster_log_destination"]
+	tokenSnapshots := input.Snapshots["token"]
+
+	if len(tokenSnapshots) == 0 {
+		input.LogEntry.Warn("log-shipper-token secret not found")
+
+		return nil
+	}
+
+	token := tokenSnapshots[0].(string)
+
+	for _, destinationSnapshot := range destinationSnapshots {
+		destination := destinationSnapshot.(v1alpha1.ClusterLogDestination)
+
+		if destination.Name == "d8-loki" {
+			continue
+		}
+
+		if destination.Spec.Type != v1alpha1.DestLoki {
+			continue
+		}
+
+		endpoint, err := url.Parse(destination.Spec.Loki.Endpoint)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse loki endpoint '%s'", destination.Spec.Loki.Endpoint)
+		}
+
+		lokiAddr, err := net.ResolveTCPAddr("tcp", endpoint.Host)
+		if err != nil {
+			return errors.Wrapf(err, "failed to resolve ip address for loki endpoint '%s'", destination.Spec.Loki.Endpoint)
+		}
+
+		deckhouseLokiAddr, err := net.ResolveTCPAddr("tcp", lokiHost)
+		if err != nil {
+			return errors.Wrapf(err, "failed to resolve ip address for loki endpoint '%s'", destination.Spec.Loki.Endpoint)
+		}
+
+		if !lokiAddr.IP.Equal(deckhouseLokiAddr.IP) || lokiAddr.Port != deckhouseLokiAddr.Port {
+			continue
+		}
```

```suggestion
		if !lokiAddr.IP.Equal(deckhouseLokiAddr.IP) || lokiAddr.Port != deckhouseLokiAddr.Port {
			continue
		}
```
if we make a HA loki in the future, this check will stop working, and this is not obvious., because it will be silently skipped.

---

done